api/src/features/todo/use_cases/delete_todo_by_id.py

```diff
@@ -1,16 +1,17 @@
 from pydantic import BaseModel, Field
+from starlette.responses import JSONResponse
 
-from common.use_case import use_case_responses
+from common.responses import create_response
 from infrastructure.repositories.TodoRepository import TodoRepository
 
 
 class DeleteTodoByIdResponse(BaseModel):
     success: bool = Field(...)
 
 
-@use_case_responses
-def delete_todo_use_case(id: str, todo_item_repository=TodoRepository()) -> DeleteTodoByIdResponse:
+@create_response(JSONResponse)
```

Comment:
Would it be better to put this in todo_feature.py? Ie
```
@router.delete("/{id}", operation_id="delete_by_id", response_model=DeleteTodoByIdResponse)
@create_response(JSONResponse)
def delete_todo_by_id(id: str):
    return delete_todo_use_case(id)
```
I would expect it to work just as well, while also separating concerns in a better manner. Though I suppose it would be harder to test that way.

Reply:
Agreed, this was just assumed not to work. As it could interfere with FastAPIs router decorator, but I just tested it, and it looks fine.